extensions/github-authentication/src/node/fetch.ts

```diff
@@ -2,6 +2,18 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-import fetch from 'node-fetch';
+import { workspace } from 'vscode';
+import fetch, { RequestInfo, RequestInit, Response } from 'node-fetch';
+import { HttpsProxyAgent } from 'https-proxy-agent';
 
-export const fetching = fetch;
+export function fetching(url: RequestInfo, init?: RequestInit | undefined): Promise<Response> {
+	const instanceOptions: RequestInit = {
+		...init
+	};
+	const httpProxy = workspace.getConfiguration('http').get('proxy', process.env.HTTP_PROXY);
+	if (!init?.agent && httpProxy) {
+		instanceOptions.agent = new HttpsProxyAgent(httpProxy);
+	}
```

Comment:
Ideally this would be reused between requests to benefit from HTTP keep-alive (would save a few hundred ms on requests after the first)